### ISSUE TITLE
Switch Canal log verbosity to warning

### ISSUE THREAD

**Issue by strowi** [rancher/rancher#19472]
**What kind of request is this (question/bug/enhancement/feature request):**
enhancement

**Steps to reproduce (least amount of steps as possible):**
Deploy cluster with rancher2 + canal via agent on baremetal.

**Result:**
Canal is spamming the logs (>=6000/30s with <15 nodes) and has 
FELIX_LOGSEVERITYSCREEN=info  set.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.1.8
- Installation option (single install/HA): HA

**Comment by superseb** [comment]
Seems to be a regression in the versioned templates, can you confirm you are on k8s 1.13?

**Comment by strowi** [comment]
```
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.5", GitCommit:"2166946f41b36dea2c4626f90a77706f426cdea2", GitTreeState:"clean", BuildDate:"2019-03-25T15:19:22Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
```
yes, i am.

**Comment by superseb** [comment]
This should patch the environment variables as they were on < 1.13 (warning, this does replace the pods)

```
kubectl -n kube-system patch ds canal -p '{"spec": {"template": {"spec": {"containers":[{"name": "calico-node", "env":[{"name":"DATASTORE_TYPE","value":"kubernetes"},{"name":"WAIT_FOR_DATASTORE","value":"true"},{"name":"NODENAME","valueFrom":{"fieldRef":{"apiVersion":"v1","fieldPath":"spec.nodeName"}}},{"name":"CALICO_NETWORKING_BACKEND","value":"none"},{"name":"CLUSTER_TYPE","value":"k8s,canal"},{"name":"FELIX_IPTABLESREFRESHINTERVAL","value":"60"},{"name":"IP"},{"name":"CALICO_IPV4POOL_CIDR","value":"192.168.0.0/16"},{"name":"CALICO_DISABLE_FILE_LOGGING","value":"true"},{"name":"FELIX_DEFAULTENDPOINTTOHOSTACTION","value":"ACCEPT"},{"name":"FELIX_IPV6SUPPORT","value":"false"},{"name":"FELIX_LOGSEVERITYSCREEN","value":"Warning"},{"name":"FELIX_HEALTHENABLED","value":"true"}]}]}}}}'
```

**Comment by superseb** [comment]
To validate Canal
- Set up 1.13 cluster with Canal
- Check configured environment variables

```
$ kubectl -n kube-system get pods -l k8s-app=canal -o go-template='{{range .items}}{{range .spec.containers}}{{ if eq .name "calico-node"}}{{"===> Pod: "}}{{.name}}{{"\n"}}{{range .env}}{{.name}}{{"="}}{{if .value}}{{.value}}{{end}}{{if .valueFrom}}{{.valueFrom}}{{end}}{{"\n"}}{{end}}{{end}}{{end}}{{end}}' | egrep "Pod|FELIX_LOGFILEPATH|FELIX_LOGSEVERITYSYS|FELIX_LOGSEVERITYSCREEN"
===> Pod: calico-node
FELIX_LOGFILEPATH=none
FELIX_LOGSEVERITYSYS=
FELIX_LOGSEVERITYSCREEN=Warning
===> Pod: calico-node
FELIX_LOGFILEPATH=none
FELIX_LOGSEVERITYSYS=
FELIX_LOGSEVERITYSCREEN=Warning
===> Pod: calico-node
FELIX_LOGFILEPATH=none
FELIX_LOGSEVERITYSYS=
FELIX_LOGSEVERITYSCREEN=Warning
```

- Check configured `log_level` in Configmap

```
kubectl -n kube-system get configmap canal-config -o go-template='{{.data.cni_network_config}}' | grep log_level
      "log_level": "WARNING",
```

- Check logline count for the containers

```
for pod in $(kubectl -n kube-system get pods -l k8s-app=canal -o custom-columns=NAME:.metadata.name --no-headers); do echo "Log count for pod $pod and container calico-node:"; kubectl -n kube-system logs $pod -c calico-node | wc -l; done
``` 

To validate Calico
- Set up 1.13 cluster with Calico
- Check configured environment variables

```
kubectl -n kube-system get pods -l k8s-app=calico-node -o go-template='{{range .items}}{{range .spec.containers}}{{"===> Pod: "}}{{.name}}{{"\n"}}{{range .env}}{{.name}}{{"="}}{{if .value}}{{.value}}{{end}}{{if .valueFrom}}{{.valueFrom}}{{end}}{{"\n"}}{{end}}{{end}}{{end}}' | egrep "Pod|FELIX_LOGFILEPATH|FELIX_LOGSEVERITYSYS|FELIX_LOGSEVERITYSCREEN"
===> Pod: calico-node
FELIX_LOGFILEPATH=none
FELIX_LOGSEVERITYSYS=
FELIX_LOGSEVERITYSCREEN=Warning
===> Pod: calico-node
FELIX_LOGFILEPATH=none
FELIX_LOGSEVERITYSYS=
FELIX_LOGSEVERITYSCREEN=Warning
===> Pod: calico-node
FELIX_LOGFILEPATH=none
FELIX_LOGSEVERITYSYS=
FELIX_LOGSEVERITYSCREEN=Warning
```

- Check configured `log_level` in Configmap

```
kubectl -n kube-system get configmap calico-config -o go-template='{{.data.cni_network_config}}' | grep log_level
      "log_level": "WARNING",
```

- Check logline count for the containers

```
for pod in $(kubectl -n kube-system get pods -l k8s-app=calico-node -o custom-columns=NAME:.metadata.name --no-headers); do echo "Log count for pod $pod and container calico-node:"; kubectl -n kube-system logs $pod -c calico-node | wc -l; done
```

**Comment by alena1108** [comment]
A note to RN the implication of the fix - the fact that calico pods will be restarted.

**Comment by jiaqiluo** [comment]
The bug is validated on Rancher: v2.2.2
- Canal
```
> kubectl -n kube-system get pods -l k8s-app=canal -o go-template='{{range .items}}{{range .spec.containers}}{{ if eq .name "calico-node"}}{{"===> Pod: "}}{{.name}}{{"\n"}}{{range .env}}{{.name}}{{"="}}{{if .value}}{{.value}}{{end}}{{if .valueFrom}}{{.valueFrom}}{{end}}{{"\n"}}{{end}}{{end}}{{end}}{{end}}' | egrep "Pod|FELIX_LOGFILEPATH|FELIX_LOGSEVERITYSYS|FELIX_LOGSEVERITYSCREEN"
===> Pod: calico-node
FELIX_LOGSEVERITYSCREEN=info
> kubectl -n kube-system get configmap canal-config -o go-template='{{.data.cni_network_config}}' | grep log_level
      "log_level": "info",
> for pod in $(kubectl -n kube-system get pods -l k8s-app=canal -o custom-columns=NAME:.metadata.name --no-headers); do echo "Log count for pod $pod and container calico-node:"; kubectl -n kube-system logs $pod -c calico-node | wc -l; done
Log count for pod canal-nhr8h and container calico-node:
1037
> 
```
- Calico
```
> kubectl -n kube-system get pods -l k8s-app=calico-node -o go-template='{{range .items}}{{range .spec.containers}}{{"===> Pod: "}}{{.name}}{{"\n"}}{{range .env}}{{.name}}{{"="}}{{if .value}}{{.value}}{{end}}{{if .valueFrom}}{{.valueFrom}}{{end}}{{"\n"}}{{end}}{{end}}{{end}}' | egrep "Pod|FELIX_LOGFILEPATH|FELIX_LOGSEVERITYSYS|FELIX_LOGSEVERITYSCREEN"
===> Pod: calico-node
FELIX_LOGSEVERITYSCREEN=info
> kubectl -n kube-system get configmap calico-config -o go-template='{{.data.cni_network_config}}' | grep log_level
      "log_level": "info",
> for pod in $(kubectl -n kube-system get pods -l k8s-app=calico-node -o custom-columns=NAME:.metadata.name --no-headers); do echo "Log count for pod $pod and container calico-node:"; kubectl -n kube-system logs $pod -c calico-node | wc -l; done
Log count for pod calico-node-2zfnn and container calico-node:
1316
> 
```


The fix is validated on Rancher: Master 5/6 ebf11403
- Canal
```
> kubectl -n kube-system get pods -l k8s-app=canal -o go-template='{{range .items}}{{range .spec.containers}}{{ if eq .name "calico-node"}}{{"===> Pod: "}}{{.name}}{{"\n"}}{{range .env}}{{.name}}{{"="}}{{if .value}}{{.value}}{{end}}{{if .valueFrom}}{{.valueFrom}}{{end}}{{"\n"}}{{end}}{{end}}{{end}}{{end}}' | egrep "Pod|FELIX_LOGFILEPATH|FELIX_LOGSEVERITYSYS|FELIX_LOGSEVERITYSCREEN"
===> Pod: calico-node
FELIX_LOGFILEPATH=none
FELIX_LOGSEVERITYSYS=
FELIX_LOGSEVERITYSCREEN=Warning
> kubectl -n kube-system get configmap canal-config -o go-template='{{.data.cni_network_config}}' | grep log_level
      "log_level": "WARNING",
> for pod in $(kubectl -n kube-system get pods -l k8s-app=canal -o custom-columns=NAME:.metadata.name --no-headers); do echo "Log count for pod $pod and container calico-node:"; kubectl -n kube-system logs $pod -c calico-node | wc -l; done
Log count for pod canal-xx9g8 and container calico-node:
20
> 
```

- Calico
```
> kubectl -n kube-system get pods -l k8s-app=calico-node -o go-template='{{range .items}}{{range .spec.containers}}{{"===> Pod: "}}{{.name}}{{"\n"}}{{range .env}}{{.name}}{{"="}}{{if .value}}{{.value}}{{end}}{{if .valueFrom}}{{.valueFrom}}{{end}}{{"\n"}}{{end}}{{end}}{{end}}' | egrep "Pod|FELIX_LOGFILEPATH|FELIX_LOGSEVERITYSYS|FELIX_LOGSEVERITYSCREEN"
===> Pod: calico-node
FELIX_LOGFILEPATH=none
FELIX_LOGSEVERITYSYS=
FELIX_LOGSEVERITYSCREEN=Warning
> kubectl -n kube-system get configmap calico-config -o go-template='{{.data.cni_network_config}}' | grep log_level
      "log_level": "WARNING",
> for pod in $(kubectl -n kube-system get pods -l k8s-app=calico-node -o custom-columns=NAME:.metadata.name --no-headers); do echo "Log count for pod $pod and container calico-node:"; kubectl -n kube-system logs $pod -c calico-node | wc -l; done
Log count for pod calico-node-krkmf and container calico-node:
90
> 
```
